documentation/LifecycleManager.md

```diff
@@ -1,18 +1,43 @@
 # lifecycleManager
 
-The Lifecycle Manager is responsible for actually configuring the users system via setting handlers and launch handlers.
-It is the only component in the system that keeps state. This is done in the "session" member of the lifecycleManager
-component, and tracks what changes have been done to the system, what the original configuration of the system was and
-which user is currently logged in.
+The Lifecycle Manager is responsible for actually performing user login, logout, retrieving the active GPII key
+requests. It also configs the users system via setting handlers and launch handlers.
 
-## LifecycleManager Queue
+The Lifecycle Manager is the only component in the system that keeps state. This is done in the "session" member of
+the lifecycleManager component, and tracks what changes have been done to the system, what the original configuration
+of the system was and which user is currently logged in.
 
-The lifecycleManager queue is used to hold the high-level action that needs to happen: starting the login process,
-starting logout process, starting the update process. Since the steps of these processes are asynchronous, the queue was
-implemented to avoid racing issues between these processes (e.g. if a logout attempt is started before login is
-complete, etc).
+## LifecycleManager User Logon Request Queue
 
-Each item in the queue should have the following format:
+The lifecycleManager user logon request queue is used to hold all requests for user logging in, logging out and
+retrieving the active GPII key. The queue is handled sequentially to avoid racing issues (e.g. the next request will
+wait until the previous request finishes before the processing).
+
+The logging in and logging out item in the queue should have the following format:
+
+```snippet
+{
+    gpiiKey: <String>,     // A GPII key
+    logonState: <String>   // "login" or "logout"
+}
+```
+
+The active GPII key request item in the queue should have the following format:
```

Comment:
I don't think this needs to be a queued action - the user might well be interested in knowing what the logon state is *now* rather than at some time in the future when the system has stopped doing things.